src/czml3/types.py

```diff
@@ -32,6 +32,90 @@ def format_datetime_like(dt_object):
     return result
 
 
+class RgbafValue(BaseCZMLObject):
+    """A color specified as an array of color components [Red, Green, Blue, Alpha]
+     where each component is in the range 0.0-1.0. If the array has four elements,
+    the color is constant. If it has five or more elements, they are time-tagged
+    samples arranged as [Time, Red, Green, Blue, Alpha, Time, Red, Green, Blue, Alpha, ...],
+    where Time is an ISO 8601 date and time string or seconds since epoch.
+
+    """
+
+    def __init__(self, *, values):
+        if not (len(values) == 4 or len(values) % 5 == 0):
+            raise ValueError(
+                "Input values must have either 4 or N * 5 values, "
+                "where N is the number of time-tagged samples."
+            )
+
+        if len(values) == 4:
+            if not all([0 <= val <= 1 for val in values]):
+                raise ValueError("Color values must be floats in the range 0-1.")
+
+        else:
+            for i in range(0, len(values), 5):
+                t, v = values[i], values[i + 1 : i + 5]
+
+                if type(t) is not float and type(t) is not int:
```

Comment:
I think the validation of the times should be in a different place perhaps? There are other properties using it

Reply:
In particular, `Position`